### PR TITLE
Edit remediation for use_approved_ciphers

### PR DESCRIPTION
**Pull request by cyarbrough76**
#### Description:

- _Added RHEL6 remediation to shared.sh rather than having separate rhel6.sh_

#### Rationale:

- _The separate rhel6 remediation script shouldnt be necessary in this case._

### CODE REVIEW

**Comment by mildas** [comment]
**Changes identified:**
Rule rhel6:
 Bash remediation was deleted. No test for it will be selected.

**Comment by jan-cerny** [comment]
Hi @cyarbrough76 

Due to https://github.com/ComplianceAsCode/content/blob/b37a12fdd8593c8f3975b23bc9445a980f61f107/linux_os/guide/services/ssh/sshd_approved_ciphers.var#L15

this will mean adding the `rijndael-cbc@lysator.liu.se` cipher suite to RHEL 6 profiles.

Is that expected?

**Comment by mildas** [comment]
**Changes identified:**
Rule rhel6:
 Bash remediation was deleted. No test for it will be selected.
Profile stig on rhel6:
 Rule  added to stig profile.

**Recommended tests to execute:**
 build_product rhel6
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel6-ds.xml stig

**Comment by cyarbrough76** [comment]
> 
> 
> Hi @cyarbrough76
> 
> Due to
> 
> https://github.com/ComplianceAsCode/content/blob/b37a12fdd8593c8f3975b23bc9445a980f61f107/linux_os/guide/services/ssh/sshd_approved_ciphers.var#L15
> 
> this will mean adding the `rijndael-cbc@lysator.liu.se` cipher suite to RHEL 6 profiles.
> 
> Is that expected?

My intention was to set the stig ciphers for use with stig profile. I have now included the var setting in the rhel6 stig profile to use the stig settings. I'm not sure if the ciphers need to change for other applications (default .var settings).

**Comment by matejak** [comment]
This PR reduces the list of approved ciphers to `aes128-ctr,aes192-ctr,aes256-ctr`, which may be unexpected, but I like the approach of this PR.

**Comment by matejak** [comment]
ok to test

**Comment by mildas** [comment]
**Changes identified:**
Rule rhel6:
 Bash remediation was deleted. No test for it will be selected.
Profile stig on rhel6:
 Rule  added to stig profile.

**Recommended tests to execute:**
 build_product rhel6
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel6-ds.xml stig

**Comment by mildas** [comment]
**Changes identified:**
Rule rhel6:
 Bash remediation was deleted. No test for it will be selected.
Profile stig on rhel6:
 Rule  added to stig profile.

**Recommended tests to execute:**
 build_product rhel6
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel6-ds.xml stig

**Comment by matejak** [comment]
I have introduced jinja to variable definitions, so if somebody uses the stig selector in profiles of a product that doesn't handle the list of ciphers/macs explicitly, they will get an appropriate build-time error.

**Comment by redhatrises** [comment]
Please undo the Jinja changes. They aren't needed as the ciphers are the same across 6 & 7.

**Comment by mildas** [comment]
**Changes identified:**
Rule rhel6:
 Bash remediation was deleted. No test for it will be selected.
Profile stig on rhel6:
 Rule  added to stig profile.

**Recommended tests to execute:**
 build_product rhel6
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel6-ds.xml stig

**Comment by matejak** [comment]
I have streamlined the macs variable, as requested.

**Comment by redhatrises** [comment]
> I have streamlined the macs variable, as requested.
> Jinja is still needed there, as it is the only way to prevent using the `stig` selector with products s.a. Fedora or Debian, i.e. products that either don't have STIG, or they have, but the project is not reflecting it.

They aren't needed, and content wise, we shouldn't be doing this. Please remove them for this PR to be merged. There is no need to prevent others from using a cipher selection based off of the STIG.

**Comment by matejak** [comment]
AFAIK, there is no such thing as one size fits all STIG - there is STIG for RHEL6, for RHEL7 etc. Same rules that are selected in STIG profiles for different products even have different STIG IDs.
You are right that in the case of macs, there is only one value which is the same for RHEL6 and RHEL7, but that's because for RHEL8, this setting is done by Crypto Policies. Trying to use `sshd_approved_macs=stig` in RHEL8 should result in compile-time error, as the list of approved ciphers exists, but is different and part of the Crypto Policies package, not part of this project.
I can imagine that if there is a new STIG for Debian, Debian maintainers should edit files in question and explicitly validate that the actual list of ciphers/macs is correct. And they won't miss it if they get a compile-time error.

**Comment by matejak** [comment]
@redhatrises please tell us your opinion on using `sshd_approved_ciphers=stig` on products that don't have STIG.

**Comment by redhatrises** [comment]
It should 100% be allowed. There should be nothing in the build system to prevent variables from being used in this way. Just because there is no STIG for a product doesn't mean that I cannot use the variables.

**Comment by mildas** [comment]
**Changes identified:**
Rule rhel6:
 Bash remediation was deleted. No test for it will be selected.
Profile stig on rhel6:
 Rule  added to stig profile.

**Recommended tests to execute:**
 build_product rhel6
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel6-ds.xml stig

**Comment by matejak** [comment]
I see a similarity with e.g. a C compiler - that one also allows to deference null pointers. Is this what you meant?

I remove the macro for that case, as the possible inappropriate usage of the `stig` variable is not something of our concern.

File: linux_os/guide/services/ssh/sshd_approved_ciphers.var [hunk 1]
```diff
@@ -11,5 +11,10 @@ operator: equals
 interactive: false
 
 options:
+    {{#- STIG is product-dependent, so the selector has to be made available for individual products selectively -#}}
+    {{%- if product in ("rhel7", "ol7") %}}
     stig: aes128-ctr,aes192-ctr,aes256-ctr
+    {{%- elif product in ("rhel6", "ol6") %}}
+    stig: aes128-ctr,aes192-ctr,aes256-ctr,aes128-cbc,3des-cbc,aes192-cbc,aes256-cbc
+    {{%- endif %}}
```
Comment (redhatrises) [inline linux_os/guide/services/ssh/sshd_approved_ciphers.var:19]:
```suggestion
    el7stig: aes128-ctr,aes192-ctr,aes256-ctr
    el6stig: aes128-ctr,aes192-ctr,aes256-ctr,aes128-cbc,3des-cbc,aes192-cbc,aes256-cbc
```

Variables need to be user friendly. They don't need to have a build selector for individual products. Selectors bread complexity and are generally not useful for variables.

Comment (matejak) [reply linux_os/guide/services/ssh/sshd_approved_ciphers.var:19]: I don't understand what you mean by "Variables ... don't need to have a build selector for individual products" while you suggest to add two build selectors for two classes of products. Could you please go into greater detail?
Regarding your approach, I generally wouldn't be against it, but in this case, it breaks backward compatibility - we use the `stig` selector for this variable in the RHEL7 STIG profile file.

Comment (redhatrises) [reply linux_os/guide/services/ssh/sshd_approved_ciphers.var:19]: > Regarding your approach, I generally wouldn't be against it, but in this case, it breaks backward compatibility - we use the stig selector for this variable in the RHEL7 STIG profile file.

Then use the following:

```
    stig: aes128-ctr,aes192-ctr,aes256-ctr
    el6stig: aes128-ctr,aes192-ctr,aes256-ctr,aes128-cbc,3des-cbc,aes192-cbc,aes256-cbc
```

Technically, the RHEL6 content will be considered EOL in November anyway.

**Comment by redhatrises** [comment]
Requested changes haven't been made. In the next two weeks, the RHEL6 removal PR will be opened since RHEL6 is EOL. Closing.

**Pull request closed without merging**